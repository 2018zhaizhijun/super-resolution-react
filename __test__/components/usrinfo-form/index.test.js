import React from 'react';
import UsrInfoForm from '@components/usrinfo-form';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Map } from 'immutable';
// import MockAdapter from 'axios-mock-adapter';
// import axiosInstance from '@service/request';
// import toJson from "enzyme-to-json";

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)
//const axiosMock = new MockAdapter(axiosInstance)

describe('User info form render', () => {
    const store = mockStore(Map({
        login: {
            profile: {
                avatarUrl: null,
                usrname: 'zz',
                telephone: '12345678910',
                sex: 1,
            }
        }
    }));

    it('info displaying state should match snapshot', () => {
        const infoWrapper = renderer.create(
            <Provider store={store}>
                <UsrInfoForm isEditing={false} />
            </Provider>
        ).toJSON();
        expect(infoWrapper).toMatchSnapshot();
    });

    it('editing state should match snapshot', () => {
        const editWrapper = renderer.create(
            <Provider store={store}>
                <UsrInfoForm isEditing={true} />
            </Provider>
        ).toJSON();
        expect(editWrapper).toMatchSnapshot();
    });

    it('should change editing state when edit button clicked', () => {
        // const mockCommitData = {
        //     'statusCode': 200,
        //     'errmsg': '成功',
        //     'data': {}
        //   }
        // axiosMock.onPost('/user/info').reply(200, mockCommitData)

        const wrapper = mount(
            <Provider store={store}>
                <UsrInfoForm isEditing={false} />
            </Provider>
        );
        expect(wrapper.exists('.infoForm')).toBeTruthy();
        
        wrapper.find('#editButton').first().simulate('click');
        expect(wrapper.exists('.editForm')).toBeTruthy();
        expect(wrapper.exists('.infoForm')).toBeFalsy();

        wrapper.find('#cancelButton').first().simulate('click');
        expect(wrapper.exists('.infoForm')).toBeTruthy();
        expect(wrapper.exists('.editForm')).toBeFalsy();

        // wrapper.find('#editButton').first().simulate('click');
        // //表单提交的回调函数是异步函数，这里无法等到其执行完再进行expect
        // wrapper.find('form').simulate('submit');
        // //expect(toJson(wrapper)).toMatchSnapshot();
        // expect(wrapper.exists('.infoForm')).toBeTruthy();
        // expect(wrapper.exists('.editForm')).toBeFalsy();
    });

})