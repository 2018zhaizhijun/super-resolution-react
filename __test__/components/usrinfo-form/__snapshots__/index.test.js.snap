// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User info form render editing state should match snapshot 1`] = `
<div>
  <form
    className="ant-form ant-form-horizontal editForm"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
        style={Object {}}
      >
        <label
          className=""
          title="用户头像"
        >
          用户头像
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-10 ant-col-sm-offset-1"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <img
              alt="avatar"
              className="avatar"
              height="120"
              src={Object {}}
            />
            <span
              className=""
            >
              <div
                className="ant-upload ant-upload-select ant-upload-select-text"
              >
                <span
                  className="ant-upload"
                  onClick={[Function]}
                  onDragOver={[Function]}
                  onDrop={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  <input
                    accept=""
                    multiple={false}
                    onChange={[Function]}
                    onClick={[Function]}
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                    type="file"
                  />
                  <button
                    className="ant-btn"
                    onClick={[Function]}
                    type="button"
                  >
                    <span
                      aria-label="upload"
                      className="anticon anticon-upload"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="upload"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 00-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                        />
                      </svg>
                    </span>
                    <span>
                      点击上传
                    </span>
                  </button>
                </span>
              </div>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item ant-form-item-has-success"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
        style={Object {}}
      >
        <label
          className=""
          htmlFor="usrname"
          title="用户名"
        >
          用户名
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-10 ant-col-sm-offset-1"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <input
              className="ant-input"
              id="usrname"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              type="text"
              value="zz"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
        style={Object {}}
      >
        <label
          className=""
          title="手机号码"
        >
          手机号码
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-10 ant-col-sm-offset-1"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <input
              className="ant-input ant-input-disabled"
              disabled={true}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              type="text"
              value="12345678910"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item ant-form-item-has-success"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
        style={Object {}}
      >
        <label
          className=""
          htmlFor="sex"
          title="性别"
        >
          性别
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-10 ant-col-sm-offset-1"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <div
              className="ant-radio-group ant-radio-group-solid"
              id="sex"
            >
              <label
                className="ant-radio-wrapper"
              >
                <span
                  className="ant-radio"
                  style={Object {}}
                >
                  <input
                    checked={false}
                    className="ant-radio-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyPress={[Function]}
                    onKeyUp={[Function]}
                    type="radio"
                    value="0"
                  />
                  <span
                    className="ant-radio-inner"
                  />
                </span>
                <span>
                  男
                </span>
              </label>
              <label
                className="ant-radio-wrapper"
              >
                <span
                  className="ant-radio"
                  style={Object {}}
                >
                  <input
                    checked={false}
                    className="ant-radio-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyPress={[Function]}
                    onKeyUp={[Function]}
                    type="radio"
                    value="1"
                  />
                  <span
                    className="ant-radio-inner"
                  />
                </span>
                <span>
                  女
                </span>
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-10 ant-col-sm-offset-9"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <button
              className="ant-btn ant-btn-primary"
              id="cancelButton"
              onClick={[Function]}
              style={
                Object {
                  "marginRight": "20px",
                }
              }
              type="button"
            >
              <span>
                取 消
              </span>
            </button>
            <button
              className="ant-btn ant-btn-primary"
              id="saveButton"
              onClick={[Function]}
              type="submit"
            >
              <span>
                保 存
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`User info form render info displaying state should match snapshot 1`] = `
<div>
  <form
    className="ant-form ant-form-horizontal infoForm"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
        style={Object {}}
      >
        <label
          className=""
          title="用户头像"
        >
          用户头像
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-10 ant-col-sm-offset-1"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <img
              alt="avatar"
              className="avatar"
              height="120"
              src={Object {}}
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
        style={Object {}}
      >
        <label
          className=""
          title="用户名"
        >
          用户名
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-10 ant-col-sm-offset-1"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <div>
              zz
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
        style={Object {}}
      >
        <label
          className=""
          title="手机号码"
        >
          手机号码
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-10 ant-col-sm-offset-1"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <div>
              12345678910
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
        style={Object {}}
      >
        <label
          className=""
          title="性别"
        >
          性别
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-10 ant-col-sm-offset-1"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <div>
              女
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-8 ant-col-sm-offset-9"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <button
              className="ant-btn ant-btn-block"
              id="editButton"
              onClick={[Function]}
              type="button"
            >
              <span
                aria-label="edit"
                className="anticon anticon-edit"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="edit"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                  />
                </svg>
              </span>
              <span>
                编辑个人信息
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
